include/tc/core/cuda/cuda_tc_executor.h

```diff
@@ -80,14 +95,14 @@ class CudaTcExecutor : public ::tc::TcExecutor {
       const std::vector<const void*>& inputs,
       const std::vector<void*>& outputs) const;
 
-  bool hasRTCFun() {
+  bool hasRuntimeCompiledFun() override {
```

Comment:
agreed with @ftynse on this one, cognitively, I also found it better to see `hasRTCFunction` so that we can relate things better to what's going on :)

Reply:
The rationale here is that whatever the backend (CPU, GPU, OpenCL, ROCm etc) there is a concept of a function that needs to be compiled at runtime before it can be evaluated. 

It happens to be called RTC / NVRTC for CUDA

Similarly to CUDA the thinking is that it should also be cleaned up when used in the autotuner.

I can't argue with the fact that this looks like premature optimization.
However the goal of this whole PR is to refactor things such that we are ready for multiplebackends.

So if y'all feel strongly about this I'll revert but I'd like this decision to be made in the context I mentioned above please :)

Reply:
We need to think of a better name.  "runtime" and "compiled" in the same phrase is non-obvious :)  For LLVM, the produced result is called a module IIRC.  Can do this later.

Reply:
@ftynse agreed but what does RTC stand for ;) ? Care to take this to NVIDIA ? :D :D 
But generally yes, let's find a better name for the abstraction and it happens to be called RTC in CUDA-land.

Reply:
RTC stands for Real-Time Clock, or réseau téléphonique commuté :P
Even if I take it somewhere, the answer is likely a bacwards-compatibility argument: it has been called this way, let's not change.